bindings/typescript/src/wrap/types.ts

```diff
@@ -0,0 +1,415 @@
+export interface Tiktoken_Module_Args_createCustomBpe {
+  encoder: Map<Types.String, Types.UInt>;
+  specialTokensEncoder: Map<Types.String, Types.UInt>;
+  pattern: Types.String;
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_encodeOrdinary {
+  tokenizer: Types.Tiktoken_Tokenizer;
+  text: Types.String;
+  bpe?: Array<Types.UInt8> | null;
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_encode {
+  tokenizer: Types.Tiktoken_Tokenizer;
+  text: Types.String;
+  allowedSpecial: Array<Types.String>;
+  bpe?: Array<Types.UInt8> | null;
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_encodeWithSpecialTokens {
+  tokenizer: Types.Tiktoken_Tokenizer;
+  text: Types.String;
+  bpe?: Array<Types.UInt8> | null;
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_decode {
+  tokenizer: Types.Tiktoken_Tokenizer;
+  tokens: Array<Types.UInt>;
+  bpe?: Array<Types.UInt8> | null;
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_splitByToken {
+  tokenizer: Types.Tiktoken_Tokenizer;
+  text: Types.String;
+  useSpecialTokens: Types.Boolean;
+  bpe?: Array<Types.UInt8> | null;
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_splitByTokenOrdinary {
+  tokenizer: Types.Tiktoken_Tokenizer;
+  text: Types.String;
+  bpe?: Array<Types.UInt8> | null;
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_r50kBase {
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_p50kBase {
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_p50kEdit {
+}
+
+/* URI: "fs/../wrap/build" */
+export interface Tiktoken_Module_Args_cl100kBase {
+}
+
+/* URI: "fs/../wrap/build" */
+export class Tiktoken {
+  protected _defaultClient: CoreClient;
+  protected _defaultUri: string;
+  protected _defaultEnv?: Record<string, unknown>;
+
+  constructor(
+    client?: CoreClient,
+    env?: Record<string, unknown>,
+    uri?: string,
+  ) {
+    this._defaultClient = this._getClient(client);
+    this._defaultEnv = this._getEnv(env);
+    this._defaultUri = this._getUri(uri);
+  }
+
+  public get client(): CoreClient {
+    return this._defaultClient;
+  }
+
+  public get uri(): string {
+    return this._defaultUri;
+  }
+
+  public get env(): Record<string, unknown> | undefined {
+    return this._defaultEnv;
+  }
+
+  private _getClient(client?: CoreClient): CoreClient {
+    return client || this._defaultClient || this._getDefaultClient();
+  }
+
+  private _getUri(uri?: string): string {
+    return uri || this._defaultUri || this._getDefaultUri();
+  }
+
+  private _getEnv(env?: Record<string, unknown>): Record<string, unknown> | undefined {
+    return env || this._defaultEnv || this._getDefaultEnv();
+  }
+
+  protected _getDefaultClient(): CoreClient {
+    return new PolywrapClient();
+  }
+  protected _getDefaultUri(): string {
+    return "fs/../wrap/build";
```

Comment:
This isn't a good default URI to have... hmm not sure how to best deal with this. Some options:
1. Deploy to wrapscan and use something like `wrapscan.io/polywrap/tiktoken@1`
2. Embed the wasm bytes, and use a URI like `embed/tiktoken` or something like that.

Reply:
i would deploy to wrapscan; and we can embed after if we'd like to